src/irmin-pack/pack_value.ml

```diff
@@ -137,6 +137,16 @@ struct
 
   let decode_bin_length = get_dynamic_sizer_exn value
   let kind _ = kind
+
+  (* we assume that computing the size of blobs is not expensive.
+
+     For "simple" blobs, we divide the blob size by a random constant
+     (100 here) to give a higher weight to large blobs in the LRU.
+
+     For "complex" blobs, we use a "large" constant to fill the LRU
+     faster. *)
+  let weight v =
+    match size_of v with None -> 100_000 | Some v -> max 1 (v / 100)
```

Comment:
I'm not sure we want to cap the weight - if someone add 1T blob it should be not added to the LRU.

Reply:
> I think it would be reasonable to require the irmin-pack users to provide a contents that has a size_of operation implemented. This would allow you to test at the module level that the size_of function is not unknown (using Repr.Size.of_value).

I am not sure to understand - tezos blobs have a dynamic size (they are strings) so `Size.of_value` will return `Dynamic`. What would you gain by using `Type.Size.of_value` instead of the derived `size_of` here?

Reply:
One point of clarification: I think the previous suggestion by @Ngoguey42 was to cap the _entry size_ allowed into the LRU, not the weight calculation size, so that items greater than a certain weight (eg `cap / 10`) would not be allowed into the LRU. The `min v (lru_size / 10)` code snippet belongs in the LRU code and would reject, for example, 1T blobs (unless of course the weight cap is >= 10T :upside_down_face:).

I'm not sure about the discussion with regards to `size_of` vs `of_value` but following along for insights.

Reply:
Got it - I've pushed a change to reflect this, let me know what you think.